src/client/common/utils/async.ts

```diff
@@ -237,3 +237,32 @@ export async function flattenIterator<T>(iterator: AsyncIterator<T, void>): Prom
     }
     return results;
 }
+
+export class ChainedExecutions<T> {
+    private pendingCellExecution?: Promise<T> | undefined;
+    public clear() {
```

Comment:
Not used, this is a WIP (trying to figure out if i need this or ensure `chainExecution` never caches & ends up in a weird state